src/frontend/src/pages/WorkPackageDetailPage/WorkPackageEditContainer/WorkPackageEditContainer.tsx

```diff
@@ -3,223 +3,211 @@
+  const {
+    fields: dependencies,
+    append: appendDependency,
+    remove: removeDependency
+  } = useFieldArray({ control, name: 'dependencies' });
 
-  const deliverablesUtil: EditableTextInputListUtils = {
-    add: (val) => {
-      const clone = dels.slice();
-      clone.push({
-        id: -1,
-        detail: val,
-        isResolved: false
-      });
-      setDels(clone);
-    },
-    remove: (idx) => {
-      const clone = dels.slice();
-      clone.splice(idx, 1);
-      setDels(clone);
-    },
-    update: (idx, val) => {
-      const clone = dels.slice();
-      clone[idx].detail = val;
-      setDels(clone);
-    }
-  };
+  const { mutateAsync } = useEditWorkPackage(workPackage.wbsNum);
 
-  const setters = {
-    setProjectLead,
-    setProjectManager,
-    setName,
-    setCrId,
-    setStartDate,
-    setDuration,
-    setDeps,
-    setEa,
-    setDels,
-    setStatus
-  };
+  if (allUsers.isLoading || !allUsers.data || !auth.user) return <LoadingIndicator />;
+  if (allUsers.isError) {
+    return <ErrorPage message={allUsers.error?.message} />;
+  }
+
+  const users = allUsers.data.filter((u) => u.role !== 'GUEST');
 
   const transformDate = (date: Date) => {
     const month = date.getUTCMonth() + 1 < 10 ? `0${date.getUTCMonth() + 1}` : (date.getUTCMonth() + 1).toString();
     const day = date.getUTCDate() < 10 ? `0${date.getUTCDate()}` : date.getUTCDate().toString();
     return `${date.getUTCFullYear().toString()}-${month}-${day}`;
   };
 
-  const transformWbsNum = (wbsNum: WbsNumber) => {
-    return {
-      carNumber: wbsNum.carNumber,
-      projectNumber: wbsNum.projectNumber,
-      workPackageNumber: wbsNum.workPackageNumber
-    };
-  };
-
-  const handleSubmit = async (event: SyntheticEvent) => {
-    event.preventDefault();
-
-    const { userId } = auth.user!;
+  const onSubmit = async (data: any) => {
+    const { name, projectLeadId, projectManagerId, startDate, duration, status } = data;
+    const expectedActivities = mapBulletsToPayload(data.expectedActivities);
+    const deliverables = mapBulletsToPayload(data.deliverables);
+    const dependencies = mapWBSNumToPayLoad(data.dependencies);
 
     const payload = {
-      projectLead,
-      projectManager,
+      projectLeadId,
+      projectManagerId,
       workPackageId: workPackage.id,
       userId,
       name,
-      crId: parseInt(crId),
+      crId: parseInt(data.crId),
       startDate: transformDate(startDate),
       duration,
-      dependencies: deps.map((dep) => transformWbsNum(dep)),
-      expectedActivities: ea,
-      deliverables: dels,
+      dependencies: dependencies,
+      expectedActivities: expectedActivities,
+      deliverables: deliverables,
       wbsElementStatus: status
     };
 
-    console.log(payload);
-
     try {
       await mutateAsync(payload);
       exitEditMode();
-    } catch (_) {
-      return;
+    } catch (e) {
+      if (e instanceof Error) {
+        alert(e.message);
```

Comment:
theres a popup when it fails that syas like "something went wrong but it was probably invalid crId" and idk where that is but we should get rid of it because now there are two alerts that happen